### PR TITLE
OOP school library: add Person, Student and Teacher classes

### PR DESCRIPTION
**Pull request by CesarHerr**
- Create class Person with the following:

- [x] Instance vars: @id, @name, and @age.
- [x] Constructor with name, age, and parent_permission as parameter. name and parent_permission are optional and have default values of "Unknown" and true.
- [x] Getters for @id, @name, and @age.
- [x] Setters for @name and @age.
- [x] Private method of_age? that returns true if @age is greater or equal to 18 and false otherwise.
- [x] Public method can_use_services? that returns true if person is of age or if they have permission from parents.

- Create class Student with the following:

- [x] Inherits from Person.
- [x] Constructor extends parent's constructor by adding @classroom and a parameter for it.
- [x] Method play_hooky that returns "¯\(ツ)/¯".

- Create class Teacher with the following:

- [x] Inherits from Person.
- [x] Constructor extends parent's constructor by adding @specialization and a parameter for it.
- [x] Override can_use_services? so it always returns true.
- [x] Each class should be saved in a separate file. :point_down:  :point_down: 
image
image
image

### CODE REVIEW

File: class_teacher.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+require_relative 'class_person'
+
+class Teacher < Person
+  attr_accessor :specialization
+
+  def initialize(age, name:, parent_permission:, specialization:)
+    super(age, name: name, parent_permission: parent_permission)
+    @specialization = specialization
+  end
```
Comment (Olamarx) [inline class_teacher.rb]: - [ ] I see you are defaulting the argument in the super, when you pass arguments to super, you're providing values for the parameters of the superclass's constructor, so the value should be inside the initialize. Apply this to the student class too, please.

**Review by Olamarx — CHANGES_REQUESTED** [review]
Hi @CesarHerr, 

Good job so far, indeed you have spent time on this project!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
- Readme file professional ✔️
- Linters passed ✔️
- All methods in place ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: class_teacher.rb (hunk 1 above)
Comment (CesarHerr) [reply class_teacher.rb]: @Olamarx Thanks for your review, I'll check it out.

**Review by thecodechaser — APPROVED** [review]
## STATUS: APPROVED 🟢

## Hi @CesarHerr👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 🎉
- Good job with classes. ✔️
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**